### ISSUE TITLE
io_interrupt example fails with asyncio

### ISSUE THREAD

**Issue by danielballan** [NSLS-II/caproto#233]
Using the new `--async-lib` argument to all IOC examples, I tested the examples and trio and asyncio for the first time. All except one passed. The failure is easy to reproduce --- details below.

I have xfailed it for now. The fix may be simple, but I'm just sidelining this to focus on other things.

<details>

```
$ python -m caproto.ioc_examples.io_interrupt --async-lib asyncio
[I 16:16:33.653 server:183] Server starting up on 0.0.0.0:5066
[I 16:16:33.653 server:103] Server starting up...
[I 16:16:33.655 server:154] Server startup complete.
* keypress method called at server startup
[E 16:16:33.656 server:173] Server error. Will shut down
    Traceback (most recent call last):
      File "/Users/dallan/Documents/Repos/caproto/caproto/asyncio/server.py", line 159, in run
        await asyncio.gather(*tasks)
      File "/Users/dallan/Documents/Repos/caproto/caproto/server/server.py", line 108, in _server_startup
        return await self.startup(self, async_lib)
      File "/Users/dallan/Documents/Repos/caproto/caproto/ioc_examples/io_interrupt.py", line 68, in keypress
        queue = async_lib.ThreadsafeQueue()
    TypeError: 'NoneType' object is not callable
[I 16:16:33.659 server:177] Server exiting....
Traceback (most recent call last):
  File "/Users/dallan/miniconda3/envs/dd/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/dallan/miniconda3/envs/dd/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/dallan/Documents/Repos/caproto/caproto/ioc_examples/io_interrupt.py", line 92, in <module>
    run(ioc.pvdb, **run_options)
  File "/Users/dallan/Documents/Repos/caproto/caproto/server/__init__.py", line 11, in run
    return run(pvdb, **kwargs)
  File "/Users/dallan/Documents/Repos/caproto/caproto/asyncio/server.py", line 198, in run
    start_server(pvdb, bind_addr=bind_addr, log_pv_names=log_pv_names))
  File "/Users/dallan/miniconda3/envs/dd/lib/python3.6/asyncio/base_events.py", line 467, in run_until_complete
    return future.result()
  File "/Users/dallan/Documents/Repos/caproto/caproto/asyncio/server.py", line 185, in start_server
    ret = await ctx.run(log_pv_names=log_pv_names)
  File "/Users/dallan/Documents/Repos/caproto/caproto/asyncio/server.py", line 159, in run
    await asyncio.gather(*tasks)
  File "/Users/dallan/Documents/Repos/caproto/caproto/server/server.py", line 108, in _server_startup
    return await self.startup(self, async_lib)
  File "/Users/dallan/Documents/Repos/caproto/caproto/ioc_examples/io_interrupt.py", line 68, in keypress
    queue = async_lib.ThreadsafeQueue()
TypeError: 'NoneType' object is not callable
```

</details>

**Comment by tacaswell** [comment]
As @klauer said else where, the asyncio backend needs to have a thread-safe queue implemented.

**Comment by danielballan** [comment]
Makes sense. Sorry for lazy bug report, frying bigger fish.

**Comment by tacaswell** [comment]
You can do this via the file descriptor functionality https://docs.python.org/3/library/asyncio-eventloop.html#watch-file-descriptors

**Comment by danielballan** [comment]
Closed by #278
